### PR TITLE
Hide peer_is_optional and origin_is_optional from docs for later deprecation

### PR DESCRIPTION
**Pull request by yangminzhu**
istio/proxy#1892

### CODE REVIEW

**Comment by yangminzhu** [comment]
@diemtvu Could you take a look? Thank you.

File: authentication/v1alpha1/policy.proto [hunk 1]
```diff
@@ -274,11 +243,12 @@ message Policy {
...
   // They will be evaluated in order; the first validate one will be used to
   // set peer identity (source.user) and other peer attributes. If none of
-  // these methods pass, and peer_is_optional flag is false (see below),
-  // request will be rejected with authentication failed error (401).
+  // these methods pass, request will be rejected with authentication failed error (401).
   // Leave the list empty if peer authentication is not required
   repeated PeerAuthenticationMethod peers = 2;
 
+  // $hide_from_docs
+  // WILL BE DEPRECATED
```
Comment (diemtvu) [inline authentication/v1alpha1/policy.proto]: You can aslo decorate the field is deprecated
```
bool peer_is_optional = 3 [deprecated = true];
```

Comment (yangminzhu) [reply authentication/v1alpha1/policy.proto]: Done

**Pull request merged**